docker/Dockerfile

```diff
@@ -4,7 +4,7 @@ FROM ubuntu:18.04
 ARG release=latest
 
 # Install required packages
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y install openssl ca-certificates curl wget && rm -rf /var/lib/apt/lists/*
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y install make cmake openssl ca-certificates curl wget && rm -rf /var/lib/apt/lists/*
```

Comment:
```suggestion
RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -y install make cmake g++ openssl ca-certificates curl wget && rm -rf /var/lib/apt/lists/*
```

So currently, there is a small problem that prevents the current PR to be merged as-is. Indeed, if there is no C++ compiler installed within the Docker image, CMake will complains with the following error:

```
-- Detecting C compile features - done
CMake Error at CMakeLists.txt:2 (project):
  No CMAKE_CXX_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
```

It seems CMake still forces the detection of a working C++ compiler. I tried customizing the C++ compiler to use the bundled cross-compiling one (the C++ to WASM one). It can be done simply be moving the `find_package(eosio.cdt)` line at the top of the CMakeFile.

One could think it would solve the problem, however CMake now tries to perform the C++ compiler checks which seems to consist in compiling a sample program. But it uses a standard program which the cross-compiling compiler chokes on.

I tried setting the CMake platform to generic via `set(CMAKE_SYSTEM_NAME "Generic")` but that did not help. I tried to find how to disable or change the CMake tests, but did not find anything.

So if somehow with more knowledge of CMake could shim in, that would be much appreciated.

Right now, the simple "workaround" is to add `g++` to the list of installed package along `cmake` and `make`.  If you think it's the best way to go to start, I will update my PR to add the package.